### ISSUE TITLE
JSC_NEGATING_A_NON_NUMBER_ERROR not reported consistantly

### ISSUE THREAD

**Issue by blickly** [google/closure-compiler#197]
This issue was imported from Closure Compiler's previous home at http://closure-compiler.googlecode.com

The original discussion is archived at:
http://blickly.github.io/closure-compiler-issues/#1158

**Comment by Dominator008** [comment]
```js
-new Array(2, String);
```
is incorrectly folded to
```js
-[2, String];
```
while
```js
-new Array(2, 'String');
```
is correctly folded to
```js
NaN
```
